### ISSUE TITLE
cppcheck cleanup

### ISSUE THREAD

**Issue by mprobson** [UIUC-PPL/charm#7]
*Original issue: https://charm.cs.illinois.edu/redmine/issues/7*

---
Removing errors detailed by cppcheck for proposal

[cppcheck1.59.out](https://github.com/UIUC-PPL/charm/files/3536154/cppcheck1.59.out.log)
[cppcheck1.61.err](https://github.com/UIUC-PPL/charm/files/3536155/cppcheck1.61.err.log)
[cppcheck1.62.err.all](https://github.com/UIUC-PPL/charm/files/3536156/cppcheck1.62.err.all.log)

**Comment by PhilMiller** [comment]
*Original date: 2013-05-04 21:11:31*

---
With the very latest cppcheck, we get 193 reports. Thankfully, relatively few of them are in core components. Most are in libraries or more peripheral pieces of the common code.

**Comment by PhilMiller** [comment]
*Original date: 2013-08-22 04:23:29*

---
From version 1.61

**Comment by nikhil-jain** [comment]
*Original date: 2013-10-14 03:47:02*

---
Fixed issues related to init.h, isomalloc, ampi etc.

**Comment by harshithamenon** [comment]
*Original date: 2013-10-14 03:51:26*

---
Fixed issues in ck-ldb.

**Comment by pplimport** [comment]
*Original author: Abhishek Gupta*
*Original date: 2013-10-17 19:24:42*

---

Fixed issues in Charmrun

**Comment by xiangni** [comment]
*Original date: 2013-10-17 21:16:32*

---
Fixed issues in FT.

**Comment by mprobson** [comment]
*Original date: 2013-10-21 21:37:28*

---
A run with cppcheck 1.62 I count 110 errors (some of which are "style" or "warning" and aren't as critical to fix but are nonetheless reported by the tool). Please take a look at the up-to-date list (I pulled before running this) and see if there are still problems in your modules.

EDIT: Fixed arch/cell/cell_lib/rawToProj.cpp error

**Comment by PhilMiller** [comment]
*Original date: 2013-10-21 23:30:20*

---
I've reviewed the isomalloc and memory.c reports, and concluded that they are false positives, though the tool wouldn't have been able to tell that.

I've fixed one of the charmxi errors even though it's not strictly a runtime component, and concluded the other isn't a problem.

I've fixed the semi-real uninitialized variable error in MSA. Cppcheck doesn't recognize that some variables are 'output' variables from PUP, so it flags them.

**Comment by PhilMiller** [comment]
*Original date: 2013-10-23 04:05:26*

---
I added suppressions for a few reports that we've concluded are false positives. Look for `cppcheck-suppress` in the code for how to record it. Note that it must appear on the line immediately preceding the one that cppcheck flags. To get the error ID to use after the keyword, run `cppcheck --xml` to get detailed output. To enable the suppressions when re-running cppcheck, pass it the `--inline-suppr` flag.
